src/main/java/com/exactpro/th2/fix/client/util/MessageUtil.java

```diff
@@ -38,40 +38,26 @@ public static MessageGroupBatch toBatch(byte[] byteArray, ConnectionID connectio
 
     public static SessionID getSessionID(String message) {
 
-        String senderSubID = "";
-        String senderLocationID = "";
-        String targetSubID = "";
-        String targetLocationID = "";
         String sessionQualifier = "";
+        return new SessionID(message.substring(2, message.indexOf('\u0001')),
+                getTagValue(message, 49), getTagValue(message, 50),
+                getTagValue(message, 142), getTagValue(message, 56),
+                getTagValue(message, 57), getTagValue(message, 143), sessionQualifier);
```

Comment:
Use QuickFIX/J constants instead of magic numbers. E.g. `SenderCompID.FIELD` instead of 49